src/main/java/com/internship/backend/service/ReservationService.java

```diff
@@ -36,79 +41,87 @@ public List<Reservation> getAllReservations(){
-        }
         return reservation;
 
     }
 
-    public Reservation update(int reservationId, Reservation newReservation){
-        Reservation reservation = reservationRepository.findById(reservationId).orElseThrow(()->new EntityNotFoundException("Reservation not found"));
+    public Reservation update(int reservationId, Reservation newReservation) throws ReservationAlreadyExists {
+        Reservation reservation = reservationRepository.findById(reservationId).orElseThrow(()->new ReservationAlreadyExists("Reservation not found"));
 
         reservation.setStartTime(newReservation.getStartTime());
         reservation.setEndTime(newReservation.getEndTime());
-
-//        reservation.setYear(reservation.getYear());
-//        reservation.setMonth(reservation.getMonth());
-//        reservation.setDay(reservation.getDay());
-//        reservation.setStartHour(reservation.getStartHour());
-//        reservation.setEndHour(reservation.getEndHour());
-
         if(reservationRepository.count()==0){
             reservationRepository.resetAutoIncrementId();
         }
 
         return reservationRepository.save(reservation);
     }
 
-    public void delete(int reservationId){
+    public void delete(int reservationId) throws ReservationDoesNotExistException {
         if(!reservationRepository.existsById(reservationId))
-            throw new RuntimeException("Reservation not found");
+            throw new ReservationDoesNotExistException("Reservation not found");
         priceRepository.deleteById(reservationId);
     }
 
-   /* public List<Integer> getReservationsByDate(int month, int day, int startHour, int endHour){
 
-        List<Reservation> reservations = reservationRepository.findAll();
-        List<Integer> tennisCourts = new ArrayList<>();
-        for(Reservation reservation: reservations){
 
-            if(reservation.getMonth() == reservations.get(month).getMonth() &&
-                    reservation.getDay() == reservations.get(day).getDay() &&
-                    reservation.getStartHour() == reservations.get(startHour).getStartHour()
-                    && reservation.getEndHour() == reservations.get(endHour).getEndHour()){
+    //metoda in care user-ul sa vada toate terenurile disponibile dintr-o anumita data
+   public List<TennisCourt> getAvailableTennisCourts(NewDate startDate, NewDate endDate){
+        int tennisCourtId;
+        List<TennisCourt> tennisCourts = tennisCourtRepository.findAll();
+        List<Reservation> reservations = reservationRepository.findAll();
 
-                tennisCourts.add(reservation.getId());
+        for (Reservation elem : reservations){
+            if(( elem.getStartTime().equals(startDate) &&  elem.getEndTime().equals(endDate))
+                    || (elem.getStartTime().isBefore(endDate) && endDate.isBefore(elem.getEndTime()))
+                    || (elem.getStartTime().isBefore(startDate) && startDate.isBefore(elem.getEndTime())))
```

Comment:
shouldn't this logic check if a court is fully booked during the date range from startDate to endDate?

Reply:
The method checks if there are reservations in the interval [startDate, newDate], if so, then it removes the reserved tennis courts and displays the available ones.

Reply:
this method checks if there is a reservation in that interval and then deletes the court, not if it is fully booked in that interval. also, shouldn't a reservation be by the hour? can a reservation be made for a month?

Reply:
For this method, there is no concept of fully or partially booked tennis court yet.
Each tennis court is only capable of having a single reservation. The method will return all tennis courts that don't have any reservation.

And the condition if(( elem.getStartTime().equals(startDate) && elem.getEndTime().equals(endDate))
simulates the cases when a user wants to make a reservation in the same time interval as another existing reservation 
I will be uploading the updated version with constraints.